server/polar/order/service.py

```diff
@@ -261,6 +261,13 @@ def __init__(self, order: Order) -> None:
         super().__init__(message)
 
 
+class PaymentAlreadyInProgress(OrderError):
+    def __init__(self, order: Order) -> None:
+        self.order = order
+        message = f"Payment for order {order.id} is already in progress"
+        super().__init__(message)
```

Comment:
Tip: you can set a status code as second argument of the error class. It's handy because we have a FastAPI exception handler automatically catching our errors and build a response accordingly. By default, the status code is 500, but in this case, since it's a known error, it probably can be a 409.

Reply:
Nice!